src/sympc/protocol/falcon/falcon.py

```diff
@@ -117,6 +119,8 @@ def mul_master(x: MPCTensor, y: MPCTensor, session: Session) -> MPCTensor:
                 z_shares, x.session
             )
             result = MPCTensor(shares=reshared_shares, session=x.session)
+            result.shape = MPCTensor._get_shape("mul", x.shape, y.shape)  # for prrs
```

Comment:
MAYBE: You reassign the result on the next line. I think this line would have no effect.

Reply:
Shape is required for prrs generation, when I pass result to `ABY3.truncate`,it infers it from result.shape
https://github.com/OpenMined/SyMPC/blob/f38d35bb6d52e6453948c9e25896bd7ae81da1db/src/sympc/protocol/falcon/falcon.py#L122

Reply:
Ahh..you are right. I missed where you use result as an argument.